File: RecordMusicCunoc/src/main/java/com/adolfosc/modelo/conexion/Solicitud.java
package com.adolfosc.modelo.conexion;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author hectoradolfo
 */
public class Solicitud {
    
    private String tipo;
    private String nombre;
    private List<Dato> datos;

    public Solicitud() {
        datos = new ArrayList<>();
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public List<Dato> getDatos() {
        return datos;
    }

    public void setDatos(List<Dato> datos) {
        this.datos = datos;
    }
    
    public void insertarDato(Dato dato){
        this.datos.add(dato);
    }
    
}

File: RecordMusicApp/app/src/main/java/com/adolfosc/modelo/conexion/Solicitud.kt
package com.adolfosc.modelo.conexion

class Solicitud {

    private var tipo: String? = null
    private var nombre: String? = null
    private var datos: MutableList<Dato>? = null

    constructor() {
        datos = ArrayList()
    }
    fun Solicitud() {
        datos = ArrayList()
    }

    fun getTipo(): String? {
        return tipo
    }

    fun setTipo(tipo: String?) {
        this.tipo = tipo
    }

    fun getNombre(): String? {
        return nombre
    }

    fun setNombre(nombre: String?) {
        this.nombre = nombre
    }

    fun getDatos(): List<Dato>? {
        return datos
    }

    fun setDatos(datos: MutableList<Dato>?) {
        this.datos = datos
    }

    fun insertarDato(dato: Dato) {
        datos!!.add(dato)
    }
}